server/src/routes/api/v1/auth/index.ts

```diff
@@ -1,17 +1,24 @@
 import { Router, Request, Response, NextFunction } from "express";
 import { AuthService } from "./../../../../logic/services/auth.service";
 import { ErrorUtils } from "./../../../../logic/utils/errorUtils";
-import { ApiError, DbError, SocoboUser } from "./../../../../models/index";
+import { 
+  ApiError, DbError, SocoboUser, LoginResult 
+} from "./../../../../models/index";
 
 
 export class AuthRouteV1 {
   constructor (private _authService: AuthService, private _router: Router) {}
 
   createRoutes (): Router {
     // login the user
-    this._router.post("/login", (req: Request, res: Response, next: NextFunction) => {
-      this._authService.login(req.body.email, req.body.password)
-        .then((result: SocoboUser) => res.status(200).json(result))
+    this._router.post("/login", this.checkLoginRequest, (req: Request, res: Response, next: NextFunction) => {
+      // get data from request
+      let isEmailLogin: boolean = req.body.isEmailLogin;
```

Comment:
Maybe we could do the isEmailOrUsername check in the backend. The frontend could just send one string in the body (email or username) and the backend would then first assume that the string is a email (check against email) if this fails the backend checks the provided string against the username and if this fails the login was invalid.

So we could avoid these logic in the frontend. Just an idea.